### PR TITLE
TopicSelectionStrategy on KafkaAvroBaseTransformer

### PR DESCRIPTION
**Pull request by pjagielski**

### CODE REVIEW

**Review by arkadius — APPROVED** [review]

File: engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/TopicSelectionStrategy.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package pl.touk.nussknacker.engine.avro
+
+import cats.data.Validated
+import pl.touk.nussknacker.engine.avro.schemaregistry.{SchemaRegistryClient, SchemaRegistryError}
+
+import java.util.regex.Pattern
+
+trait TopicSelectionStrategy {
+
+  val schemaRegistryClient: SchemaRegistryClient
```
Comment (mproch) [inline engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/TopicSelectionStrategy.scala:10]: I think this is not needed here? Let's either remove it, or put schemaRegistryClient to getTopics parameter (and make Strategy stateless)

File: engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/KafkaAvroBaseTransformer.scala [hunk 2]
```diff
@@ -25,14 +25,16 @@ trait KafkaAvroBaseTransformer[T] extends SingleInputGenericNodeTransformation[T
 
   def schemaRegistryProvider: SchemaRegistryProvider
 
+  def topicSelectionStrategy(schemaRegistryClient: SchemaRegistryClient): TopicSelectionStrategy = new AllTopicsSelectionStrategy(schemaRegistryClient)
```
Comment (mproch) [inline engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/KafkaAvroBaseTransformer.scala:28]: If we make it def topicSelectionStrategy: TopicSelectionStrategy,
and pass schemaRegistryClient as parameter to strategy method, we'll enable having strategy as field in subclasses

Comment (arkadius) [reply engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/KafkaAvroBaseTransformer.scala:28]: woops good point :+1: , I forgotten that this `getTopicParam` is invoken quite often.

Comment (pjagielski) [reply engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/KafkaAvroBaseTransformer.scala]: moved as param

**Review by mproch — APPROVED** [review]

**Pull request merged**